lib/lightning_web/live/credential_live/credential_index_component.ex

```diff
@@ -98,34 +158,40 @@ defmodule LightningWeb.CredentialLive.CredentialIndexComponent do
            project_oauth_clients: project_oauth_clients
          }
        )}
+    else
+      not_authorized(socket)
     end
   end
 
   def handle_event("edit_oauth_client", %{"id" => client_id}, socket) do
     %{oauth_clients: oauth_clients} = socket.assigns
     client = Enum.find(oauth_clients, fn client -> client.id == client_id end)
 
-    with :ok <- can_edit_credential(socket, client) do
+    if can_edit_credential(socket.assigns.current_user, client) do
       {:noreply,
        assign(socket,
          active_modal: :edit_oauth_client,
          credential: nil,
          oauth_client: client
        )}
+    else
+      not_authorized(socket)
```

Comment:
🤩

Reply:
And yes to adding common reply patterns, when doing controller responses, there are ton of wrappers for conn we use without thinking, we should at least have those two from that post, and a not_authorized one - but yeah I know why we haven't tried very hard on this so far; it's easy to find exceptions to the rule, but yeah paying attention to the exceptions and trying to mold them back into common patterns should be the goal instead.